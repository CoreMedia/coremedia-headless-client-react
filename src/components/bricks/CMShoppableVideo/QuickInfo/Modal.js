// @flow
import PropTypes from 'prop-types';
import styled from 'styled-components';

import ReactModalAdapter from './ReactModalAdapter';
import { media } from '../../../../styles/themes/utils';

const Modal = styled(ReactModalAdapter).attrs({
  overlayClassName: {
    base: 'cm-overlay',
    afterOpen: 'cm-overlay--after-open',
    beforeClose: 'cm-overlay--before-close',
  },
  modalClassName: {
    base: 'cm-modal',
    afterOpen: 'cm-modal--after-open',
    beforeClose: 'cm-modal--before-close',
  },
  closeTimeoutMS: 200,
})`
  .cm-overlay {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: rgba(173, 173, 173, 0.9);
    cursor: pointer;
    z-index: 99999;
    opacity: 0;
    &--after-open {
      opacity: 1;
      transition: opacity 150ms ease-out;
    }
    &--before-close {
      opacity: 0;
      transition: opacity 150ms ease-in;
    }
  }
  .cm-modal {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: #fff;
    box-sizing: content-box;
    cursor: default;
    z-index: 99999;
    transform: scale(0.5);
    &--after-open {
      transform: scale(1);
      transition: all 150ms ease-out;
    }
    &--before-close {
      transform: scale(0.5);
      transition: all 150ms ease-in;
    }
    ${props => media(props.theme.breakpoints.tablet)`
      top: 50%;
      left: 50%;
      width: 550px;
      height: auto;
      min-height: 365px;
      transform: translate(-50%, -50%) scale(0.5);
      box-shadow: 0px 0px 7px 0px rgba(0, 0, 0, 0.32);
      &--after-open {
        transform: translate(-50%, -50%) scale(1);
        transition: all 150ms ease-out;
      }
      &--before-close {
        transform: translate(-50%, -50%) scale(0.5);
        transition: all 150ms ease-in;
      }
    `};
  }
`;

Modal.displayName = 'Modal';

Modal.propTypes = {
  theme: PropTypes.shape({
    breakpoints: PropTypes.shape({
      tablet: PropTypes.number.isRequired,
    }).isRequired,
  }),
};

Modal.defaultProps = {
  theme: {
    breakpoints: {
      tablet: 768,
    },
  },
};

export default Modal;
