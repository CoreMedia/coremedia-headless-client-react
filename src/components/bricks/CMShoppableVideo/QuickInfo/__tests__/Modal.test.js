// @flow
import React from 'react';
import 'jest-styled-components';

import Modal from '../Modal';

describe('Modal Component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Modal />);
    expect(wrapper).toMatchSnapshot();
  });
});
