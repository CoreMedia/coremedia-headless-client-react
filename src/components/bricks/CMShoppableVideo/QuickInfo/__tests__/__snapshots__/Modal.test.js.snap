// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component should render correctly 1`] = `
.c0 .cm-overlay {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(173,173,173,0.9);
  cursor: pointer;
  z-index: 99999;
  opacity: 0;
}

.c0 .cm-overlay--after-open {
  opacity: 1;
  -webkit-transition: opacity 150ms ease-out;
  transition: opacity 150ms ease-out;
}

.c0 .cm-overlay--before-close {
  opacity: 0;
  -webkit-transition: opacity 150ms ease-in;
  transition: opacity 150ms ease-in;
}

.c0 .cm-modal {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: #fff;
  box-sizing: content-box;
  cursor: default;
  z-index: 99999;
  -webkit-transform: scale(0.5);
  -ms-transform: scale(0.5);
  transform: scale(0.5);
}

.c0 .cm-modal--after-open {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transition: all 150ms ease-out;
  transition: all 150ms ease-out;
}

.c0 .cm-modal--before-close {
  -webkit-transform: scale(0.5);
  -ms-transform: scale(0.5);
  transform: scale(0.5);
  -webkit-transition: all 150ms ease-in;
  transition: all 150ms ease-in;
}

@media (min-width:48em) {
  .c0 .cm-modal {
    top: 50%;
    left: 50%;
    width: 550px;
    height: auto;
    min-height: 365px;
    -webkit-transform: translate(-50%,-50%) scale(0.5);
    -ms-transform: translate(-50%,-50%) scale(0.5);
    transform: translate(-50%,-50%) scale(0.5);
    box-shadow: 0px 0px 7px 0px rgba(0,0,0,0.32);
  }

  .c0 .cm-modal--after-open {
    -webkit-transform: translate(-50%,-50%) scale(1);
    -ms-transform: translate(-50%,-50%) scale(1);
    transform: translate(-50%,-50%) scale(1);
    -webkit-transition: all 150ms ease-out;
    transition: all 150ms ease-out;
  }

  .c0 .cm-modal--before-close {
    -webkit-transform: translate(-50%,-50%) scale(0.5);
    -ms-transform: translate(-50%,-50%) scale(0.5);
    transform: translate(-50%,-50%) scale(0.5);
    -webkit-transition: all 150ms ease-in;
    transition: all 150ms ease-in;
  }
}

<ReactModalAdapter
  className="c0"
  closeTimeoutMS={200}
  modalClassName={
    Object {
      "afterOpen": "cm-modal--after-open",
      "base": "cm-modal",
      "beforeClose": "cm-modal--before-close",
    }
  }
  overlayClassName={
    Object {
      "afterOpen": "cm-overlay--after-open",
      "base": "cm-overlay",
      "beforeClose": "cm-overlay--before-close",
    }
  }
  theme={
    Object {
      "breakpoints": Object {
        "tablet": 768,
      },
    }
  }
/>
`;
